openeo_processes_dask/process_implementations/cubes/general.py

```diff
@@ -12,9 +12,13 @@
 
 def drop_dimension(data: RasterCube, name: str) -> RasterCube:
     if name not in data.dims:
-        raise DimensionNotAvailable()
+        raise DimensionNotAvailable(
+            f"Provided dimension ({name}) not found in data.dims: {data.dims}"
+        )
     if len(data[name]) > 1:
-        raise DimensionLabelCountMismatch()
+        raise DimensionLabelCountMismatch(
+            "The number of dimension labels exceeds one, which requires a reducer."
```

Comment:
We could make this error message more informative as well, like:
f"The number of dimension labels exceeds one, which requires a reducer. Dimension ({name}) has {len(data[name])} labels."